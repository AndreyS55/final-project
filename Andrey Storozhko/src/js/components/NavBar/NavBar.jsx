import React from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
import styles from './NavBar.scss';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  handleOpen = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  };

  render() {
    const { isOpen } = this.state;
    return (
      <React.Fragment>
        <nav className={classNames(styles.headerNav, { [styles.navIsOpen]: isOpen })}>
          <NavLink
            exact to="/"
            className={classNames(styles.navAll, { [styles.responsive]: isOpen })}
            activeClassName={styles.navActive}
            onClick={isOpen ? this.handleOpen : null}
          >
            All pokemons
          </NavLink>
          <NavLink
            to="/caught"
            className={classNames(styles.navCaught, { [styles.responsive]: isOpen })}
            activeClassName={styles.navActive}
            onClick={isOpen ? this.handleOpen : null}
          >
            Caught pokemons
          </NavLink>
        </nav>
        <button className={styles.navIcon} onClick={this.handleOpen}>&#9776;</button>
      </React.Fragment>
    );
  }
}

export default NavBar;
