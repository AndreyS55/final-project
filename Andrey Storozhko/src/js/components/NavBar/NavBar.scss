@import '../../../styles/environment/colors';
@import '../../../styles/environment/mixins';

.headerNav {
  height: 100%;
  display: flex;
  flex-direction: row;
  align-items: stretch;
  color: $wht;

  @include tablets() {
    display: none;
  }

  @include phones() {
    display: none;
  }
}

.navIsOpen {
  @include tablets() {
    display: block;
    width: 100%;
  }

  @include phones() {
    display: block;
    width: 100%;
  }
}

.navAll {
  padding: 10px;
  display: flex;
  align-items: center;
  background-image: linear-gradient(to top, $blu 0%, $blu 1%, $blu 100%);

  &:hover {
    background-color: rgba(137, 255, 253, 0.2);
  }

  @include tablets() {
    display: block;
    width: 100vw;
    padding: 15px;
  }

  @include phones() {
    display: block;
    width: 100vw;
    padding: 15px;
  }
}

.navCaught {
  padding: 10px;
  display: flex;
  align-items: center;
  background-image: linear-gradient(to top, $blu 0%, $blu 1%, $blu 100%);

  &:hover {
    background-color: rgba(137, 255, 253, 0.2);
  }

  @include tablets() {
    display: block;
    width: 100vw;
    padding: 15px;
  }

  @include phones() {
    display: block;
    width: 100vw;
    padding: 15px;
  }
}

.navActive {
  box-shadow: inset 0 0 160px -20px rgba(0,0,0,0.3);
}

.navIcon {
  border: none;
  background: none;
  color: $wht;
  align-self: center;
  display: none;
  font-size: 2rem;

  @include tablets() {
    position: absolute;
    display: block;
    top: 18px;
    right: 25px;
  }

  @include phones() {
    position: absolute;
    display: flex;
    top: 18px;
    right: 25px;
  }
}

.responsive {
   position: relative;
   top: 69px;
   left: -203px;
}
