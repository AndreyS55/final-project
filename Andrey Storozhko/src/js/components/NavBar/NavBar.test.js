import React from 'react';
import NavBar from './NavBar';

it('should render a NavBar', () => {
  const wrapper = shallow(<NavBar />);
  expect(wrapper).toMatchSnapshot();
});
