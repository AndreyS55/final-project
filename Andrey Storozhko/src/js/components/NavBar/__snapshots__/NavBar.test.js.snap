// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a NavBar 1`] = `
<Fragment>
  <nav
    className="headerNav"
  >
    <NavLink
      activeClassName="navActive"
      aria-current="page"
      className="navAll"
      exact={true}
      onClick={null}
      to="/"
    >
      All pokemons
    </NavLink>
    <NavLink
      activeClassName="navActive"
      aria-current="page"
      className="navCaught"
      onClick={null}
      to="/caught"
    >
      Caught pokemons
    </NavLink>
  </nav>
  <button
    className="navIcon"
    onClick={[Function]}
  >
    â˜°
  </button>
</Fragment>
`;
